### ISSUE TITLE
Clarify the IDP API response when the user is not signed in.

### ISSUE THREAD

**Issue by dj2** [w3c-fedid/FedCM#218]
If the user is not signed into the IDP, what does the API endpoint return (an HTTP error? An empty account list?)

**Comment by npm1** [comment]
I believe we treat any the same way, but might still be good to document best practices

**Comment by jacobbogers** [comment]
> So how do i know if i should use "active" beforehand ???

So, I guess the RP can set a first party cookie   User Agent <--> R, absent this marker always

- `setStatus('logged-out')`
- use mode `active`

like this?

**Comment by jacobbogers** [comment]
> You need to set the login status in the IDP, not the RP...

There is no misunderstanding about this, either use a header or `navigator.login.setStatus`

My previous question, how does the RP KNOW the login status? the RP is not telepathic and `navigator.login.setStatus` is write only, so

> whereas passive mode can be used in other scenarios.

Ok, this scenario:

1. The user arrives at RP,  
2. the user does not have an account  at the IDP, 
3. So how does the code running on the RP know that it needs to set `mode:active` in this case?

**Comment by jacobbogers** [comment]
>If the user does not have an account with the IDP, why would they want to login to the RP with the IDP?

Fair enough... but why is there a context value `signup` (as supposed to sign-in, continue, etc)
 
signup means the user has no account with the idp

Scenario 2
- The user does have an account with the idp
- The user visits the rp for the first time.

Should the code on the Rp try to sign in with mode passive .. and if that fails present the "sign in with idp" button?

**Comment by jacobbogers** [comment]
> In general, FedCM is meant to be used to login to the RP with an existing IDP account. I wouldn't recommend gating showing a login to IDP button on passive mode failing since the rejection sent to the RP is often delayed (for privacy reasons).

Thank you, I get it about the "sign in with <idp>" button should indeed use `mode: active`.

What would I do in this  scenario 

- the user does have a valid login session with the IDP and the RP using "passive mode" is in process of getting the credentials
- the user is impatient and presses the "login in <IDP>" at the same time (kicking off a active mode sign in)

**Comment by npm1** [comment]
In that case, the pending passive request is rejected and the active one shows UI. This is done automatically by the browser

**Comment by hlflanagan** [comment]
I propose closing this issue for now. If the original poster, any of the commenters, or someone in the WG or CG wants to advocate for keeping it open and help move it forward, please comment within two weeks of this label being applied.
